apps/io-services-cms-webapp/src/utils/apim-proxy.ts

```diff
@@ -0,0 +1,49 @@
+import { ApiManagementClient } from "@azure/arm-apimanagement";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as TE from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+import {
+  ApimRestError,
+  getSubscription,
+  getUser,
+  getUserGroups,
+  parseOwnerIdFullPath,
+} from "../apim_client";
+import { Delegate } from "./service_review_proxy";
+
+export const apimProxy = (
+  apimClient: ApiManagementClient,
+  apimResourceGroup: string,
+  apim: string
+) => ({
+  getDelegateFromServiceId: (
+    serviceId: NonEmptyString
+  ): TE.TaskEither<ApimRestError, Delegate> =>
+    pipe(
+      getSubscription(apimClient, apimResourceGroup, apim, serviceId),
+      TE.chain((subscription) =>
+        pipe(
+          parseOwnerIdFullPath(subscription.ownerId as NonEmptyString),
+          (ownerId) => getUser(apimClient, apimResourceGroup, apim, ownerId),
+          TE.chainW((user) =>
+            pipe(
+              getUserGroups(
+                apimClient,
+                apimResourceGroup,
+                apim,
+                user.name as NonEmptyString
+              ),
+              TE.map((userGroups) => ({
+                firstName: user.firstName,
+                lastName: user.lastName,
+                email: user.email,
+                permissions: userGroups.map((group) => group.name),
+              }))
+            )
+          )
+        )
+      )
+    ),
```

Comment:
You want to avoid unnecessary pipe nesting, and aim for a flatter code structure:
```suggestion
    pipe(
      getSubscription(apimClient, apimResourceGroup, apim, serviceId),
      TE.map((subscription) =>
        parseOwnerIdFullPath(subscription.ownerId as NonEmptyString)
      ),
      TE.chain((ownerId) =>
        getUser(apimClient, apimResourceGroup, apim, ownerId)
      ),
      TE.chainW((user) =>
        pipe(
          getUserGroups(
            apimClient,
            apimResourceGroup,
            apim,
            user.name as NonEmptyString
          ),
          TE.map((userGroups) => ({
            firstName: user.firstName,
            lastName: user.lastName,
            email: user.email,
            permissions: userGroups.map((group) => group.name),
          }))
        )
      )
    ),
```

Reply:
[4c732a9](https://github.com/pagopa/io-services-cms/pull/25/commits/4c732a9a92b60bb0a847a99868385359dbbd247c)